File: solution/day23/src/test/java/org/craftedsw/tripservicekata/user/UserShould.java
package org.craftedsw.tripservicekata.user;

import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.craftedsw.tripservicekata.user.UserBuilder.aUser;
import static org.junit.jupiter.api.DisplayNameGenerator.ReplaceUnderscores;

@DisplayNameGeneration(ReplaceUnderscores.class)
class UserShould {
    private final User rick = aUser().build();
    private final User morty = aUser().build();

    @Test
    void return_false_when_2_users_are_not_friends() {
        var user = aUser()
                .friendsWith(rick)
                .build();

        assertThat(user.isFriendWith(morty))
                .isFalse();
    }

    @Test
    void return_true_when_2_users_are_friends() {
        var user = aUser()
                .friendsWith(rick, morty)
                .build();

        assertThat(user.isFriendWith(morty))
                .isTrue();
    }
}

File: solution/c#/Day23/Day23.Tests/Users/UserShould.cs
using Day23.Users;
using FluentAssertions;
using Xunit;
using static Day23.Tests.Users.UserBuilder;

namespace Day23.Tests.Users
{
    public class UserShould
    {
        private readonly User _rick = AUser().Build();
        private readonly User _morty = AUser().Build();

        [Fact]
        public void Return_False_When_2_Users_Are_Not_Friends()
            => AUser()
                .FriendsWith(_rick).Build()
                .IsFriendWith(_morty)
                .Should()
                .BeFalse();

        [Fact]
        public void Return_True_When_2_Users_Are_Friends()
            => AUser()
                .FriendsWith(_rick, _morty).Build()
                .IsFriendWith(_morty)
                .Should()
                .BeTrue();
    }
}